docs/content/docs/dev/datastream/fault-tolerance/state.md

```diff
@@ -474,9 +474,10 @@ ttl_config = StateTtlConfig \
 For more fine-grained control over some special cleanup in background, you can configure it separately as described below.
 Currently, heap state backend relies on incremental cleanup and RocksDB backend uses compaction filter for background cleanup.
 
-##### Cleanup in full snapshot
+##### Cleanup in full scan snapshot
```

Comment:
I'm not sure whether adding `scan` everywhere adds more clarity.
because it is not explained anywhere what is a "full-scan snapshot".

Reply:
That's why I explained what kind of snapshot is a full scan snapshot below.

Reply:
Do you mean this statement:
```
Additionally, you can activate the cleanup at the moment of taking the full scan state snapshot (including the canonical savepoint,
or the full/incremental checkpoint of hashmap state-backend, or the full checkpoint of RocksDB state-backend), which will reduce its size.
```
?

It only enumerates the cases, but doesn't explain the "full scan". 
And thtat's probably right, I don't think we should overload the user with extra terms and their details.
Besides that, state backends might implement it differently.

Reply:
I think it's good to show all cases here.

Reply:
I mean that **just** enumarating the cases is enough to clarify this TTL policy;
But it's not enough to explain what the "full scan snapshot" is.

So I think either the definition of "full scan snapshot" should be given; or the "full scan snapshot" term should be removed.

Removing seems preferrable lest overload users with unnecessary information.

Reply:
Hmm... we can refactor the docs later if this will really bother users.

Reply:
I'm afraid we'll never know that :)
And the term is also introduced into the API.

Reply:
I just feel that the changes would be a bit too large if dropping the term since we still have them in the doc of the previous version. Moreover, we still support this API. That's why I prefer to still leave this term there.

Reply:
By term I [mean](https://github.com/apache/flink/pull/20421#discussion_r940348087) `scan`, which is introduced by **this** PR.

And I think it's better to keep unresolved conversations unresolved, even if the PR is going to be merged in the end.

Reply:
I see. Since any changes to public API should be considered new features. I will first merge this PR today just before the feature freeze deadline. We can change related docs or even revert this change during the release testing phase.

Reply:
I'm afraid that changing the API is not allowed during the release testing phase, so it won't be possible to revert this PR.